### PR TITLE
handing in my happy thoughts project

### PR DESCRIPTION
**Pull request by HenrikeW**
Link to my deployed project: https://henrikes-happy-thoughts.netlify.app/

My reflections on how this week's project turned out: In the end I'm happy with how the project turned out but struggled a lot on the way. The concept of useEffect is new, of course, but even useState and components are still pretty new as well. I find it difficult to wrap my head around how things are connected with each other in React which makes building projects with it quite difficult. Thus, I sticked to the blue requirements this week. 

Things I'd like to have clarified or explained in more detail: Just in general, components and how to think about them.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,28 @@
...
+
+The design follows a template provided for the assignment.
+
+## Tech used
+
+I built the site with React in JavaScript, also using HTML and CSS. It was the first time I used useEffect in React and the first time I used the POST-method while communicating with an API. Further, the "moment" library for JS is used to present the time stamp for each message in a nice way to the user.
+
+## Browser compatibility
+
+The site is responsive and was tested on a desktop PC with dev tools in Chrome, Firefox and Edge, as well as on an Android phone in Chrome and Ecosia. Everything works as expected.
```
Comment (monkeybusiness101) [inline README.md:23]: You explain the problem in a good and concise way
You explain some of the components or features without going too much into detail, nice!
You explain the tech you have used and your experience with it. This is something I should do as well, GJ!
You even performed tasks outside of the requirements (cross-browser, responsive), A+

File: code/src/Footer.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+
+export const Footer = () => {
+    return (
+        <footer>
+            <p>Created by Henrike Wiemker</p>
+            <p>Technigo Bootcamp</p>
+            <p>November 2020</p>
+            <img src={require("./logo.svg")} alt="The Technigo logo"/>
```
Comment (monkeybusiness101) [inline code/src/Footer.js:9]: I had issues with adding images and now you taught me how to add a local image. Thank you!

File: code/src/ThoughtBox.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import moment from "moment";
```
Comment (monkeybusiness101) [inline code/src/ThoughtBox.js:2]: Separate these sections of dependencies with one space, as you have with the other components.

File: code/src/ThoughtBox.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+                            {"❤️"}
+                            </span>
+                        </button>
+                        <p>x {thought.hearts}</p>
+                    </div>
+                    <p>{moment(thought.createdAt).fromNow()}</p>
+                </div>
+            </div>
+    );
+};
```
Comment (monkeybusiness101) [inline code/src/ThoughtBox.js:42]: I am by far not an expert on indentation (get feedback on it every week). I seem that after your return you have several tabs while the rest has several spaces.

File: code/src/ThoughtForm.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    };
+
+    return (
+        <div className="form-box">
+            <form onSubmit={handleFormSubmit}>
+                <label htmlFor="mythought-input">
+                What's making you happy right now?
+                    <textarea 
+                        id="mythought-input"
+                        rows="3"
```
Comment (monkeybusiness101) [inline code/src/ThoughtForm.js:32]: Nice way of setting the height for the textbox!

File: code/src/ThoughtForm.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+                        value={myThought}
+                        onChange={event => setMyThought(event.target.value)}
+                    ></textarea>
```
Comment (monkeybusiness101) [inline code/src/ThoughtForm.js:35]: </textarea> could be on its own line

File: code/src/ThoughtForm.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
+import React, {useState} from "react";
+import {THOUGHTS_URL} from "./urls";
```
Comment (monkeybusiness101) [inline code/src/ThoughtForm.js:2]: Separate these two imports with one blank line

File: code/src/ThoughtBox.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+        fetch(LIKE_URL, {
+            method: "POST",
+            body: "",
+            headers: {
+                "Content-Type": "application/json"
+            },
+        })
+        .then(() => getMessages());
+    };
```
Comment (monkeybusiness101) [inline code/src/ThoughtBox.js:18]: Nice way of importing getMessages and then using it after POSTing a like 👍

File: code/src/index.js [hunk 9]
```diff
@@ -1,6 +1,6 @@
-import React from 'react'
-import ReactDOM from 'react-dom'
-import './index.css'
-import { App } from './App'
+import React from "react";
+import ReactDOM from "react-dom";
+import "./index.css";
+import {App} from "./App";
```
Comment (monkeybusiness101) [inline code/src/index.js:4]: The order should be (according to Maks)

import React from "react";
import ReactDOM from "react-dom";

import {App} from "./App";

import "./index.css";

**Review by monkeybusiness101 — APPROVED** [review]
- Your page should follow the design as closely as possible
- You should list the most recent thoughts at the top and older thoughts at the bottom (sorted)
- Your thoughts should show the content of the message and how many likes they've received
- You should have a form to post new thoughts
- You should implement the heart button to send likes on a thought

Hi Henrike!

You have completed all the blue requirements and more! GJ!! The site looks great. You even taught me a thing or two.

I gave you some small comments on how to separate the import/dependencies and indentation. Feel to fix them if you have the time.

/Max

File: code/src/index.js (hunk 9 above)
Comment (EmOlsz) [reply code/src/index.js:4]: 👍

**Pull request closed without merging**